server/server.js

```diff
@@ -258,6 +259,66 @@ module.exports = function(port, db, githubAuthoriser) {
         });
     });
 
+    app.post("/api/groups", function(req, res) {
+        var groupInfo = req.body;
+        var creatorID = req.session.user;
+        groups.find({
+            name: groupInfo.name
+        }).limit(1).next().then(function(group) {
+            if (!group) {
+                groups.insertOne({
+                    name: groupInfo.name,
+                    description: groupInfo.description,
+                    users: [creatorID]
+                }).then(function(result) {
+                    res.json(cleanIdField(result.ops[0]));
+                }).catch(function(err) {
+                    res.sendStatus(500);
```

Comment:
Just realised, I'm not sure if you need this catch here? I think the wrapping catch should do the same? Test it to be sure.